### ISSUE TITLE
darwin-rebuild switch errors with: submoduleWith called with unexpected argument 'description'

### ISSUE THREAD

**Issue by afh** [LnL7/nix-darwin#482]
After a recent update of my nix channels (i.e. `nix channel --update`) `darwin-rebuild switch` fails with the following error message:

```sh
% darwin-rebuild switch
building the system configuration...
error: 'submoduleWith' at /nix/store/6kq1vhlln8jk09anqv8hbi2prpmfhsd4-nixpkgs/nixpkgs/lib/types.nix:568:7 called with unexpected argument 'description'

       at $HOME/.nix-defexpr/channels/home-manager/nixos/common.nix:14:14:

           13|
           14|   hmModule = types.submoduleWith {
             |              ^
           15|     description = "Home Manager module";
(use '--show-trace' to show detailed location information)

% nix-info -m
 - system: `"aarch64-darwin"`
 - host os: `Darwin 21.6.0, macOS 12.5`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.8.0`
 - channels(root): `"nixpkgs"`
 - channels($USER): `"darwin, home-manager"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
 
% nix-channel --list
darwin https://github.com/LnL7/nix-darwin/archive/master.tar.gz
home-manager https://github.com/nix-community/home-manager/archive/master.tar.gz
 ```
 
I'm uncertain where this issue should be raised best…
Here as I'm invoking `darwin-rebuild`? [Home-manager](https://github.com/nix-community/home-manager) as it calls `submoduleWith` with an unexpected argument? Nix as the `submoduleWith` does not support the `description` parameter.

My underlying Nix (2.8.0) may be outdated, but I do not know how to upgrade it using `darwin-rebuild`; I tried `nix upgrade-nix`, yet that failed with `error: Nix on NixOS must be upgraded via 'nixos-rebuild'`.

I looked at [nix-darwin's website](http://daiderd.com/nix-darwin/) and its [manual](https://daiderd.com/nix-darwin/manual/index.html) but was unable to find info on how to upgrade the underlying Nix.

Any help is greatly appreciated.

**Comment by mlyxshi** [comment]
use `nixos-unstable` branch

**Comment by afh** [comment]
Thanks @mlyxshi, running
```sh
% nix-channel --add https://nixos.org/channels/nixpkgs-unstable
% nix-channel --update nixpkgs
```
seems to have fixed it for me.

**Comment by andreivolt** [comment]
I get the same error even after adding `nixpkgs-unstable`:

```
$ nix-channel --list
darwin https://github.com/LnL7/nix-darwin/archive/master.tar.gz
home-manager https://github.com/nix-community/home-manager/archive/master.tar.gz
nixpkgs https://nixos.org/channels/nixpkgs-unstable
```

```
$ nix-channel --update && darwin-rebuild switch
building the system configuration...
error: 'submoduleWith' at /nix/store/93c5zncvirzw8wnj62lkylfdfkf45i1x-nixpkgs/nixpkgs/lib/types.nix:568:7 called with unexpected argument 'description'
```

**Comment by andreivolt** [comment]
Solved by also adding home-manager 22.05 branch:

```
$ nix-channel --add https://github.com/nix-community/home-manager/archive/release-22.05.tar.gz home-manager
```
